### PR TITLE
Toc for Sensitive Data storage cheatsheet

### PR DESCRIPTION
**Pull request by northdpole**
Thank you for submitting a Pull Request (PR) to the Cheat Sheet Series. 

> :triangular_flag_on_post: If your PR is related to grammar/typo mistakes, please double-check the file for other mistakes in order to fix all the issues in the current cheat sheet.

Please make sure that for your contribution:

- [ ] In case of a new Cheat Sheet, you have used the [Cheat Sheet template](https://github.com/OWASP/CheatSheetSeries/blob/master/templates/New_CheatSheet.md).
- [ ] All the markdown files do not raise any validation policy violation, see the [policy](https://github.com/OWASP/CheatSheetSeries#editor--validation-policy).
- [ ] All the markdown files follow these [format rules](https://github.com/OWASP/CheatSheetSeries#conversion-rules).
- [ ] All your assets are stored in the **assets** folder.
- [ ] All the images used are in the **PNG** format.
- [ ] Any references to websites have been formatted as [TEXT](URL)
- [ ] You verified/tested the effectiveness of your contribution (e.g., the defensive code proposed is really an effective remediation? Please verify it works!).
- [ ] The CI build of your PR pass, see the build status [here](https://travis-ci.org/OWASP/CheatSheetSeries/pull_requests).

If your PR is related to an issue, please finish your PR text with the following line:

This PR covers issue #<insert number here>.

Thank you again for your contribution :smiley:

### CODE REVIEW

**Comment by rbsec** [comment]
Hi @northdpole.

My initial thoughts would be about how this fits in with the existing cheat sheets. We currently have:

* Choosing and Using Security Questions (covers how to store answers for them)
* Cryptographic storage (covers different ways to encrypt, algorithms, key management, etc)
* Database Security (covers access control/permissions for database)
* Key Management (although this is an older cheat sheet and could do with an update)
* Password Storage (covers password/API key hashing)

Looking at the proposed ToC, it feels like there would be quite a lot of duplication with the existing cheat sheets, and potentially more confusion for the end user in terms of where they should be looking for information. Should this be a new cheat sheet in its own right, or perhaps should the we fill in any missing gaps in the existing cheat sheets with this content?

@ThunderSon @mackowski, what do you think?

Thanks

**Comment by ThunderSon** [comment]
My thoughts on this are to make this pertaining to topics related to secure storage, and linking accordingly into the other ones. One example would be that this CS talks about permissions, and the DB CS references it. This can be done based on which is the bigger focus. So a KMS CS focuses on the keys and how to secure them, so this CS should reference them, and where the KMS needs things related to general storage factors, it references it from this CS

**Comment by ThunderSon** [comment]
@northdpole is busy currently, so this will be put on hold till they're around for more updates.

**Comment by northdpole** [comment]
I can write something like this, got some notes already.
Thanks for being considerate! Highly appreciated :-)

**Comment by northdpole** [comment]
> I like this but we have to very carefully make sure to not duplicate the content of other CSs. I like @ThunderSon idea how it can look like.
> But what I like even more is to cover this topic #124 this is wider topic but if @northdpole can take it with our help it will be awesome. It will also have more unique content that will justify the need of a separate CS.

We can make this the opposite, it could be as simple as pointing to subsections of other cheatsheets instead of being a full blown document. What do you think? this would minimize overall edits you have to do.

@ThunderSon 
I can write this in my current timeframe :) ( got some notes already from another project).
Thanks for being considerate! Highly appreciated :-)

In short, I'd be happy to help. Would you like a draft or otherwise how do you prefer on getting started?

**Comment by ThunderSon** [comment]
@northdpole Based on our discussion on Slack, we agreed on a small touch-up on the ToC. Once that is done, I'll move forward and link sheets to the relevant sections, and then it'd be clearer for you on what is required and missing to be described properly.

**Comment by mackowski** [comment]
@northdpole are you still around? This can be an awesome CS

**Comment by northdpole** [comment]
Yes I'm around!  Happy to create it! I'm currently on holiday which may
also accelerate stuff ;)  gimme a week or so for the first draft

**Comment by mackowski** [comment]
@northdpole good to hear that! If you need any help from us let us know :)

**Comment by northdpole** [comment]
added some content and comments, let me know what do you think and i can edit accordingly :)

File: cheatsheets_draft/Sensitive_Data_Storage_Cheat_Sheet.md
Comment (mackowski) [inline cheatsheets_draft/Sensitive_Data_Storage_Cheat_Sheet.md:33]: please update link in a whole file to use retaive paths and link to relevant parts of other cheatsheets e.g.
Cryptographic_Storage_Cheat_Sheet.md#key-management

File: cheatsheets_draft/Sensitive_Data_Storage_Cheat_Sheet.md
Comment (mackowski) [inline cheatsheets_draft/Sensitive_Data_Storage_Cheat_Sheet.md:35]: how this is different than '# Managing Secrets'? Keys are just secrets thy should my m anaged the same way as other secrets: Cryptographic_Storage_Cheat_Sheet.md#key-management

File: cheatsheets_draft/Sensitive_Data_Storage_Cheat_Sheet.md
Comment (mackowski) [inline cheatsheets_draft/Sensitive_Data_Storage_Cheat_Sheet.md:48]: I would like to do this, maybe we can start here and it this grow too much we can create new CS

File: cheatsheets_draft/Sensitive_Data_Storage_Cheat_Sheet.md
Comment (mackowski) [inline cheatsheets_draft/Sensitive_Data_Storage_Cheat_Sheet.md:30]: how it will differ than DBMS Security or File System Security? What do you mean by Cloud Storage? For me Cloud Storage is a vague term to describe all methods how you can store files in the cloud e.g. DBMS in cloud or File System in cloud. 
There is clearly more storage systems in the cloud so mabey we should be more specific here? because for DBMS in cloud we will use the same proptections as for on-prem DBMS

File: cheatsheets_draft/Sensitive_Data_Storage_Cheat_Sheet.md
Comment (mackowski) [inline cheatsheets_draft/Sensitive_Data_Storage_Cheat_Sheet.md:32]: We are also creating new CS: https://github.com/OWASP/CheatSheetSeries/issues/124

**Comment by northdpole** [comment]
For tokenizing, it is really a niche subject not directly related to sensitive data storage. It is closer to Sensitive Data Handling. So I removed it from this cheatsheet but opened a ticket to create a new one https://github.com/OWASP/CheatSheetSeries/issues/471

File: cheatsheets_draft/Sensitive_Data_Storage_Cheat_Sheet.md
Comment (northdpole) [reply cheatsheets_draft/Sensitive_Data_Storage_Cheat_Sheet.md:28]: should we postpone merging until #124 lands?

File: cheatsheets_draft/Sensitive_Data_Storage_Cheat_Sheet.md
Comment (northdpole) [reply cheatsheets_draft/Sensitive_Data_Storage_Cheat_Sheet.md]: For tokenizing, it is really a niche subject not directly related to sensitive data storage. It is closer to Sensitive Data Handling. So I removed it from this cheatsheet but opened a ticket to create a new one #471

File: cheatsheets_draft/Sensitive_Data_Storage_Cheat_Sheet.md
Comment (northdpole) [reply cheatsheets_draft/Sensitive_Data_Storage_Cheat_Sheet.md]: meant s3 and similar. but you're right it is essentially file storage

**Comment by mackowski** [comment]
@jmanico what do you think about this CS? Do we want to keep it, this is basically linking to all content from other CSs?

**Comment by jmanico** [comment]
> @jmanico what do you think about this CS? Do we want to keep it, this is basically linking to all content from other CSs?

I think this is a really interesting idea. I really want to welcome other contributions that use our sheets in unique ways. :)

**Review by jmanico — APPROVED** [review]

**Review by mackowski — CHANGES_REQUESTED** [review]
@northdpole The links are not working. Links in the Contents are good but for all other links you are using a wrong syntax. After you fix it we can merge.

**Comment by mackowski** [comment]
@northdpole do you need any help from us?

**Comment by mackowski** [comment]
@northdpole hey do you need any help from us? Do you want to work on this further?

**Comment by mackowski** [comment]
I am closing this PR because no updes here were made in about 6 months. @northdpole feel free to re-open it whan you will have time to work on that.

**Pull request closed without merging**
